src/PhpAttribute/NodeFactory/AnnotationToAttributeKeyTypeCaster.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PhpAttribute\NodeFactory;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Scalar\LNumber;
+use PhpParser\Node\Scalar\String_;
+use Rector\Php80\ValueObject\AnnotationToAttribute;
+use Rector\PhpAttribute\Enum\AnnotationClassName;
+
+final class AnnotationToAttributeKeyTypeCaster
+{
+    /**
+     * @var array<string, string[]>
+     */
+    private const INTEGER_KEYS_BY_CLASS = [
+        AnnotationClassName::LENGTH => ['min', 'max'],
+    ];
+
```

Comment:
@etshy I made a static list to cast known keys. What other keys do you think we should include here?

Reply:
I only had the case on min/max for Length constraints so I'm not sure right now.
Basically any integer available ? I guess they all could be written as string in annotation but type hinted as int as Attributes.

I'll take a look of the Constraints list after work hours and could make a list from that

Reply:
>  I guess they all could be written as string in annotation but type hinted as int as Attributes.


Good point. I'll see if we can do reflection on existing attribute constructor and infer the `int` type from those.

Reply:
I had a few minute before going back to work so here a non exhaustive list if you still need it

```
CIDR
netmaskMin
netmaskMax

PasswordStrength
minScore

NoSuspiciousCharacters
checks (there are constants in docs but it's still an int)
restrictionLevel (same)

Range
max
min

File
filenameMaxLength

Image
maxHeight
maxPixels
maxRatio (float)
maxWidth
minHeight
minPixels
minRatio (float)
minWidth

Count
divisibleBy
max
min
```

Reply:
Awesome, thanks for checking. I'll use this list to test this rule in the wild, once deployed.